[FLINK-7529] Retrieve complete REST address from gateway

tillrohrmann opened the pull request:
## What is the purpose of the change

With this change, the complete REST address (protocol://hostname:port) is retrieved
from the RestfulGateway. That way we decouple the RestHandlers from the underlying
RestServerEndpoint/WebRuntimeMonitor because they no longer have to know whether
HTTPs is enabled or not.

## Verifying this change

This change is already covered by existing tests, such as `RedirectHandlerTest`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

zentol commented on flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java:
```diff
@@ -479,41 +479,6 @@ public void testNoCopyFromJar() throws Exception {
 		}
 	}
 
-	private WebRuntimeMonitor startWebRuntimeMonitor(
```
This change belongs into the previous commit, i.e. `Create DispatcherRestEndpoint and integrate with Dispatcher`.

tillrohrmann replied:
True. Will remove this change from the PR.

tillrohrmann commented:
Thanks for the review @zentol. I've addressed your comments. Once the preceding PRs have been merged and Travis gives green light, I'll merge this PR.
